### ISSUE TITLE
I am a beginner of MatSim. I read the matsim guide book install necessary software and collecting the input network and population data as well. Now I am struggling how to built my own input file and hook into the matsim and run through eclipse where I integrated the matsim. Could anyone suggest me the steps? It would be really highly appreciated.

### ISSUE THREAD

**Issue by mparvez1992** [matsim-org/matsim-code-examples#455]

**Comment by Mayoookh** [comment]
This will help you. Its step wise description as you need:
https://www.simunto.com/matsim/tutorials/eifer2019/slides_day1.pdf

**Comment by mparvez1992** [comment]
Dear,

Thank you so much for your response and materials. I am still struggling
with my issues and definitely will go through the material. If I have any
other issues I will let you know.
Thank you again.

Best regards,
Mahmud Parvez
PhD Candidate (Neuroergonomics Lab)
Department of Industrial & Systems Engineering
NC A&T State University, Greensboro, NC, USA
Phone:3364053335; Email: mparvez@aggies.ncat.edu

**Comment by mparvez1992** [comment]
Dear Mayoookh,
Thank you again for sharing that material. I am still struggling as I don't have so much basic knowledge in JAVA. Is strong JAVA knowledge is mandate for to build up the project through matsim? I tried to use some source code but these codes seems to much error from my side. Could you please guide me? should I need to learn JAVA?
I am really struggling and spending too much time but I am not able to achieve any fruitful result. 
Thanks.

**Comment by Mayoookh** [comment]
yes you do need basic knowledge of java if you have to run the basic modules of MATSim. If you read the material I have shared above, the author has explained each step of java needed for performing initial tasks. If you want to create your own module, edit network, etc. you need more than basic knowledge of JAVA

**Comment by JWJoubert** [comment]
I (strongly) agree with @Mayoookh, @mparvez1992. MATSim is a "code laboratory". Only the most basic examples, like the `equil` or the example models, will be runnable with the graphical user interface (GUI), i.e. point-and-click. For the rest, you need to plug the bits together _progamatically_. From personal experience, I recommend you first focus on Java; mainly so that you can learn to help yourself.
